Hermes/appscale/hermes/stats/producers/proxy_stats.py

```diff
@@ -266,6 +278,111 @@ def get_stats():
     client.close()
 
 
+def get_frontend_ip_port(configs_dir, proxy_name):
+  proxy_conf_path = os.path.join(configs_dir, '{}.cfg'.format(proxy_name))
+  with open(proxy_conf_path) as proxy_conf:
+    for line in proxy_conf:
+      line = line.strip()
+      if line.startswith('bind'):
+        return line.split(' ')[1].split(':')
+  raise BoundIpPortNotFound("Couldn't find bound IP and port for {} at {}"
+                            .format(proxy_name, configs_dir))
+
+
+def get_connections(ip, port):
+  return len([conn for conn in psutil.net_connections()
+             if conn.laddr == (ip, port)])
+
+
+def get_stats_from_one_haproxy(socket_path, configs_dir):
+  # Get CSV table with haproxy stats
+  csv_buf = get_stats(socket_path)
+  csv_buf.seek(2)  # Seek to the beginning but skip "# " in the first row
+  table = csv.DictReader(csv_buf, delimiter=',')
+  if ProxiesStatsSource.first_run:
+    missed = ALL_HAPROXY_FIELDS - set(table.fieldnames)
+    if missed:
+      logging.warn("HAProxy stats fields {} are missed. Old version of HAProxy "
+                   "is probably used (v1.5+ is expected)".format(list(missed)))
+    ProxiesStatsSource.first_run = False
+
+  # Parse haproxy stats output line by line
```

Comment:
I know we didn't have it before, but is there a chance to explain a bit more what we are parsing and what the parts are? Explaining a bit more BACKEND and FRONTEND and mentioning how we have service per application, and how the servers are constructed?

Reply:
@obino , if you look into definition of HAProxyFrontendStats, HAProxyBackendStats, HAProxyServerStats and HAProxyListenerStats (see below), it contains some kind of detailed list of fields related to different types of csv lines. A link to haproxy docs is also there.
And this is not a new code actually=) - github didn't recognize copy-paste which I did for supporting two haproxy processes.